### PR TITLE
Fourth question

### PR DESCRIPTION
**Pull request by Grandys**
Migration of fourth question functionality, from AWS lambda & DynamoDB to postgres.
Further steps required to finish migration:
1. Change in janusz-the-boot, to invoke janusz-backend instead of AWS lambda
2. Manual migration from DynamoDB to postgres.

Voting is not implemented in this PR.

### CODE REVIEW

File: src/main/java/com/softwaremill/janusz_backend/fourth_question/FourthQuestionController.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+import static org.springframework.util.MimeTypeUtils.APPLICATION_JSON_VALUE;
+
+@RestController
+@RequestMapping(value = "/rest/fourth-question", consumes = APPLICATION_JSON_VALUE, produces = APPLICATION_JSON_VALUE)
+@RequiredArgsConstructor
+public class FourthQuestionController {
+
+    private static Log logger = LogFactory.getLog(FourthQuestionController.class);
+    public static final String GET_FOURTH_QUESTION_URI_REGEX = "/rest(/|//)fourth-question";
```
Comment (tdziurko) [inline src/main/java/com/softwaremill/janusz_backend/fourth_question/FourthQuestionController.java]: Why this way (duplicated `//`)?

File: src/main/java/com/softwaremill/janusz_backend/fourth_question/FourthQuestionHandler.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        LocalDateTime now = nowProvider.now();
+        Question question = Question.newQuestion(
+                newQuestion.getQuestion(),
+                newQuestion.getAuthor(),
+                now);
+        fourthQuestionRepository.save(question);
+        return fourthQuestionRepository.countByAskedDateIsNull();
+    }
+
+    public Optional<TodayQuestionDto> getQuestion() {
```
[the remark this answers is not in the document]
Comment (Grandys) [reply src/main/java/com/softwaremill/janusz_backend/fourth_question/FourthQuestionHandler.java]: Renamed to `getQuestionForToday`

File: src/main/java/com/softwaremill/janusz_backend/fourth_question/FourthQuestionHandler.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        LocalDate today = getToday();
+        return fourthQuestionRepository.findByAskedDate(today);
+    }
+
+    private Optional<Question> queryOldestUnaskedQuestionAndMarksAsAsked() {
+        return fourthQuestionRepository.findFirstByAskedDateIsNullOrderByAddedTimeAsc()
+                .map(this::markAsAsked);
+    }
+
+    private Question markAsAsked(Question question) {
```
[the remark this answers is not in the document]
Comment (Grandys) [reply src/main/java/com/softwaremill/janusz_backend/fourth_question/FourthQuestionHandler.java]: Yes I am, renamed to `markAsAskedToday`

File: src/main/java/com/softwaremill/janusz_backend/fourth_question/FourthQuestionHandler.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        fourthQuestionRepository.setAskedByUUID(getToday(), question.getUuid());
+        return question;
+    }
+
+    private LocalDate getToday() {
```
[the remark this answers is not in the document]
Comment (Grandys) [reply src/main/java/com/softwaremill/janusz_backend/fourth_question/FourthQuestionHandler.java]: Ok, done

File: src/test/groovy/com/softwaremill/janusz_backend/fourth_question/GetFourthQuestionRequestHandlerSpec.groovy [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+                        .content("How was your last weekend?")
+                        .author("Mark")
+                        .build()
+        ).forEach { question -> repository.save(question) }
+        when:
+        def question = handler.getQuestion()
+        then:
+        question.isPresent()
+        def questionResponse = question.get()
+        questionResponse.question == "How was your weekend?"
```
[the remark this answers is not in the document]
Comment (Grandys) [reply src/test/groovy/com/softwaremill/janusz_backend/fourth_question/GetFourthQuestionRequestHandlerSpec.groovy]: Ok, fixed.

File: src/test/groovy/com/softwaremill/janusz_backend/fourth_question/AddFourthQuestionRequestHandlerSpec.groovy [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+
+    def "should save new question"() {
+        given:
+        def question = NewQuestionDto.of("How was your weekend?", "Michal")
+        when:
+        handler.saveQuestion(question)
+        then:
+        def questions = repository.findAll()
+        questions.size() == 1
+        questions.get(0).question == "How was your weekend?"
```
[the remark this answers is not in the document]
Comment (Grandys) [reply src/test/groovy/com/softwaremill/janusz_backend/fourth_question/AddFourthQuestionRequestHandlerSpec.groovy]: Ok, fixed.

File: src/main/java/com/softwaremill/janusz_backend/fourth_question/FourthQuestionController.java (hunk 1 above)
Comment (Grandys) [reply src/main/java/com/softwaremill/janusz_backend/fourth_question/FourthQuestionController.java]: Copy pastle from `KudosRestController`. I've removed duplicated slash.

**Pull request merged**